### DISCUSSION TITLE
Milvus with GPU

### DISCUSSION THREAD

**Discussion by kevinzezel** [milvus-io/milvus#7423] [Q&A and General discussion]
Please, any example on how to configure Milvus to search with GPU?

Thanks

**Comment by yhmo** [comment]
Currently the v2.0 cannot support GPU, yet.
If you are using v1.x, this documentation is for gpu mode: https://milvus.io/docs/v1.1.1/milvus_docker-gpu.md

The major steps:
1. download a milvus gpu version docker image
`sudo docker pull milvusdb/milvus:1.1.1-gpu-d061621-330cc6`

2. download a config file
`wget https://raw.githubusercontent.com/milvus-io/milvus/v1.1.1/core/conf/demo/server_config.yaml`

3. modify the config file, set gpu.enable=true
  ```
  gpu:
    enable: true
    cache.enable: false
    cache_size: 1GB
    gpu_search_threshold: 1000
    search_devices:
      - gpu0
    build_index_devices:
      - gpu0
  ```

  The "cache.enable" is to decide whether to use gpu memory as a cache. If true, milvus will try keep data in gpu memory.
  The "cache_size" is to define the cache size that milvus can hold data in gpu memory, this value should less than your gpu memory, 60% is better.
  The "gpu_search_threshold" is decide in which condition use gpu to do search. "gpu_search_threshold=1000" means use gpu only when your search nq >=1000, otherwise will use cpu to search
  The "search_devices" defines which gpu device can be used for searching. You can specify multi devices like this: "- gpu0, gpu1"
  The "build_index_devices" defines which gpu device can be used for building index. You can specify multi devices like this: "- gpu3, gpu4"

4. start milvus server, make sure use the config file to luanch:
  ```
  sudo docker run -d --name milvus_gpu_1.1.1 --gpus all \
  -p 19530:19530 \
  -p 19121:19121 \
  -v /home/$USER/milvus/db:/var/lib/milvus/db \
  -v /home/$USER/milvus/conf:/var/lib/milvus/conf \
  -v /home/$USER/milvus/logs:/var/lib/milvus/logs \
  -v /home/$USER/milvus/wal:/var/lib/milvus/wal \
  milvusdb/milvus:1.1.1-gpu-d061621-330cc6
  ```
